site-root/cbl/powertools/backfill-missing-ratings.php

```diff
@@ -1,279 +0,0 @@
-                                'FROM `%9$s` %10$s '.
-                            'GROUP BY StudentID, CompetencyID '.
-                        ') as %10$s '.
-                        'ON  %10$s.CompetencyID = %8$s.CompetencyID '.
-    
-                    'WHERE '.
-                    '%4$s.ID IN (%14$s) '.
-                    // 'AND %15$s '.
-                    'AND %10$s.StudentID = %4$s.StudentID '.
-                    'AND %13$s.SkillID NOT IN ( '.
-                            'SELECT %11$s.SkillID '.
-                                'FROM `%1$s` %11$s '.
-                                'WHERE %11$s.DemonstrationID = %4$s.DemonstrationID '.
-                    ')'.
-                ')'.
-            ") missingSkills";
-    
-        $params = [
-            DemonstrationSkill::$tableName, // 1
-            DB::escape(DemonstrationSkill::class), // 2
-    
-            StudentTask::$tableName, // 3
-            StudentTask::getTableAlias(), // 4
-    
-            TaskSkill::$tableName, // 5
-            TaskSkill::getTableAlias(), // 6
-    
-            Skill::$tableName, // 7
-            Skill::getTableAlias(), // 8
-    
-            StudentCompetency::$tableName, // 9
-            StudentCompetency::getTableAlias(), // 10
-    
-            DemonstrationSkill::getTableAlias(), // 11
-    
-            StudentTaskSkill::$tableName, // 12
-            StudentTaskSkill::getTableAlias(), // 13
-    
-            join(', ', $expiredStudentTaskIds), // 14
-    
-            Task::$tableName, // 15
-            Task::getTableAlias() // 16
-        ];
-    
-        if (!empty($_REQUEST['debug'])) {
-            $missingSkills = DB::allRecords($query, $params);
-            Debug::dump([$query, $params, vsprintf($query, $params)], false, 'query/params');
-            Debug::dump($missingSkills, true, 'missing skills');
-        }
-    
-        DB::nonQuery($insertStatement.$query, $params);
-        Debug::dump(DB::affectedRows(), false, 'created');
-    }
-
-}
-
-
-$studentsValue = empty($_REQUEST['students']) ? 'all' : $_REQUEST['students'];
-$minDateValue = empty($_REQUEST['minDate']) ? Term::getCurrent()->getMaster()->StartDate : $_REQUEST['minDate'];
-$maxDateValue = empty($_REQUEST['maxDate']) ? date('Y-m-d') : $_REQUEST['maxDate'];
-
-?>
-
-<html>
-    <body>
-        <h1>Backfill Missing Ratings</h1>
-
-        <form method="post">
-            <label>
-                Students:
-                <input type="text" name="students" value="<?=$studentsValue?>"/>
-            </label>
-            <label>
-                Min Date
-                <input type="date" name="minDate" value="<?=$minDateValue?>"/>
-            </label>
-            <label>
-                Max Date
-                <input type="date" name="maxDate" value="<?=$maxDateValue?>"/>
-            </label>
-            <button>Filter</button>
-        </form>
-    </body>
-
-</html>
```

Comment:
@BillClinton looks like this file became empty at the new location

Reply:
odd.  sorry I missed that.